runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -2178,18 +2205,41 @@ bool TR_EscapeAnalysis::checkOtherDefsOfLoopAllocation(TR::Node *useNode, Candid
       if (trace())
          traceMsg(comp(), "      Look at def node [%p] for use node [%p]\n", defNode, useNode);
 
+      // _aliasesOfAllocNode contains sym refs that are just aliases for a fresh allocation
+      // i.e. it is just a simple attempt at tracking allocations in cases such as :
+      // ...
+      // a = new A()
+      // ...
+      // b = a
+      // ...
+      // c = b
+      //
+      // In this case a, b and c will all be considered aliases of an alloc node and so a load of
+      // any of those sym refs will be treated akin to how the fresh allocation would have been in the below logic
+      //
+
       bool allnewsonrhs = false;
-      if ((_valueNumberInfo->getValueNumber(defNode) == _valueNumberInfo->getValueNumber(candidate->_node)))
+      //if ((_valueNumberInfo->getValueNumber(defNode) == _valueNumberInfo->getValueNumber(candidate->_node)))
```

Comment:
again remove dead code. Once we agree on the change a change to remove the {}s and reduce the indentation is probably also needed.

Reply:
Removed dead code and braces, and adjusted indentation in commit 414b74b10957a8ae1b8a4426f0c29dec2a905275